mautic/mautic
# Updated the stat count for week and added tests

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bugfix? | Yes
| New feature? |  No
| Automated tests included? | Yes
| Related user documentation PR URL | NA
| Related developer documentation PR URL | NA
| Issues addressed (#s or URLs) | NA
| BC breaks? | No
| Deprecations? | NA


[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Email graph shows wrong data for week time unit.

Currently, the graph visual within the email view page for certain emails is showing incorrect data when set to specific date ranges:
1. When using a date range of less than a month, the graph displays as expected
1. When adding a day past one month, the graph displays the wrong count

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Navigate to Emails
1. Open any Email
1. Update the filter values as
1.1 Select a range more than a month

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
1. Navigate to Emails
1. Open any Email
1. Update the filter values as
1.1 Select a range as a month and click on the apply button and see the graph
1.1 Select a range more than a month and click on the apply button and see the graph

#### Other areas of Mautic that may be affected by the change:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2. 

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. The StatsDAO is covered for getWeek and now 100% coverage for the same.
1. The Calculator is covered too.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10543"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
